integration/ctr_software_updatable_test.go

```diff
@@ -0,0 +1,367 @@
+			} else if event.Path == suite.constructStatusPath(things.SoftwareUpdatableFeatureID, "installedDependencies") {
+				if isRemoving {
+					isInstalledDeps = true
+					return false, nil
+				}
+				return true, nil
+			} else if event.Path == suite.constructStatusPath(things.SoftwareUpdatableFeatureID, "lastFailedOperation") {
+				eventValue, err = parseMap(event.Value)
+				require.NoError(suite.T(), err, "failed to parse event value")
+
+				propertyMessage, err := parseString(eventValue["message"])
+				require.NoError(suite.T(), err, "failed to parse property message")
+				return true, fmt.Errorf(propertyMessage)
+			}
+
+			return true, fmt.Errorf("event for an unexpected container status while removing is received")
+		}
+		return true, fmt.Errorf("unknown message is received")
+	})
+	if err != nil {
+		wsConnection.Close()
+	}
+	return err
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallRemove() {
+	ctrID, err := suite.installContainer(suite.createParameters(false))
+	require.NoError(suite.T(), err, "failed to process installing the container")
+
+	require.NoError(suite.T(), suite.removeContainer(createRemoveParams(ctrID)), "failed to process removing the container")
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallWithInvalidParameters() {
+	wsConnection := suite.createWSConnection()
+	defer suite.closeUnsubscribe(wsConnection)
+
+	params := make(map[string]interface{})
+	_, err := util.ExecuteOperation(suite.Cfg, suite.suFeatureURL, actionInstall, params)
+	if err != nil {
+		wsConnection.Close()
+	}
+	require.Errorf(suite.T(), err, "failed to execute software updatable install for containers with params %v", params)
+
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareInstallWithWrongChecksum() {
+	_, err := suite.installContainer(suite.createParameters(true))
+	require.ErrorContains(suite.T(), err, "internal runtime error")
+}
+
+func (suite *softwareUpdatableSuite) TestSoftwareRemoveNonExistingContainer() {
+	ctrID := "NonExistingContainer"
+	require.ErrorContains(suite.T(), suite.removeContainer(createRemoveParams(ctrID)), "container with ID = "+ctrID+" does not exist")
+}
+
+func (suite *softwareUpdatableSuite) createParameters(wrongChecksum bool) map[string]interface{} {
+	src, err := download(validContainerURL)
+	require.NoError(suite.T(), err, "unable to download file from url "+validContainerURL)
+	splitStr := strings.Split(validContainerURL, "/")
+	filePath := "/tmp/" + splitStr[len(splitStr)-1]
+	require.NoError(suite.T(), os.WriteFile(filePath, src, 7777), "unable to write file with path "+filePath)
+
+	defer require.NoError(suite.T(), os.Remove(filePath), "unable to remove file with path "+filePath)
+	return createInstallParams(filePath, src, wrongChecksum)
+}
+
+func createInstallParams(filePath string, src []byte, wrongChecksum bool) map[string]interface{} {
+	fileInfo, ctrStruct, err := getCtrImageStructure(filePath)
+	if err != nil {
+		return make(map[string]interface{})
+	}
```

Comment:
In case of some error related to the file operations the test will fail in a not a user friendly manner
```suggestion
func (suite *softwareUpdatableSuite) createParameters(wrongChecksum bool) map[string]interface{} {
	src, err := download(validContainerURL)
	require.NoError(suite.T(), err, fmt.Sprintf("unable to download file from url %s", validContainerURL))
	filePath := "valid.json"
	require.NoError(suite.T(), os.WriteFile(filePath, src, 0755), "unable to write file with path "+filePath)
	fileInfo, ctrStruct, err := getCtrImageStructure(filePath)
	require.NoError(suite.T(), err, "unable to create container image struct")
	defer require.NoError(suite.T(), os.Remove(filePath), "unable to remove file with path "+filePath)
	return createInstallParams(fileInfo, ctrStruct, src, wrongChecksum)
}

func createInstallParams(fileInfo fs.FileInfo, ctrStruct CtrImageStruct, src []byte, wrongChecksum bool) map[string]interface{} {
	splitStr := strings.Split(ctrStruct.Image.Name, "/")
	swModuleStr := strings.Split(splitStr[len(splitStr)-1], ":")
```